[Kaptain] MLflow (#4563)
mesosphere/dcos-docs-site: pages/dkp/kaptain/2.1.0/tutorials/mlflow/index.md

With MLflow, you are able to:

- Gather and track data related to experiments for analysis.
- Package ML/AI code and share it with other data scientists or transfer to production environments.
- Manage and deploy models from ML libraries to model serving and inference platforms.
- Manage the full lifecycle of an MLflow model from a centralized store (model versioning, stage transitions, annotations, etc.)

## Prerequisites

- You have launched a Jupyter notebook with Kaptain. (Using another notebook would require you to build a custom image for this purpose)

  <p class="message--important"><strong>IMPORTANT: </strong>If you decide to use your own custom image or an image not provided by Kaptain to run a distributed training job, create an image following the (<a href="../sdk/image-builder/">recommended container image creation steps</a>. Also ensure MLflow has been installed on it.</p>

> - Optional: If you are using SDK, you have [set up Kaptain SDK for your desired training framework](https://docs.d2iq.com/dkp/kaptain/2.0.0/tutorials/sdk/).

Author: ```suggestion
- Optional: If you are using SDK, you have [set up Kaptain SDK for your desired training framework](../sdk/).
```

>   <p class="message--important"><strong>IMPORTANT: </strong>If you decide to use your own custom image or an image not provided by Kaptain to run a distributed training job, create the image following the recommended container image creation steps (<a href="https://catalog.ngc.nvidia.com/orgs/nvidia/containers/l4t-base">Running an l4t-base container</a>, <a href="https://gitlab.com/nvidia/container-images/l4t-base">l4t-base build instructions</a>). Also ensure MLflow has been installed on it.</p>

Author: SGTM! Tbh I think I found those links while doing my research and assumed that was the correct way. But referencing our own docs about it is definitely much better!
```suggestion
  <p class="message--important"><strong>IMPORTANT: </strong>If you decide to use your own custom image or an image not provided by Kaptain to run a distributed training job, create an image following the (<a href="../sdk/image-builder/">recommended container image creation steps</a>. Also ensure MLflow has been installed on it.</p>
```

## Set up MLflow

> Starting from this release, MLflow is available in every notebook per default. You do not need to download, nor install it.

Reviewer: ```suggestion
Starting with Kaptain 2.1.0, MLflow is available in every notebook per default. You do not need to download, nor install it.
```

Author: We usually avoid using version numbers whenever we can, so we don't need to touch the pages between releases. However, you made me realize we should not use wordings like "starting from this release" either 🙈 
```suggestion
MLflow is available in every notebook per default. You do not need to download, nor install it.
```

To use Kaptain’s MLflow tracking server from an image that is not part of Kaptain’s distribution, nor a child-image of one, the image should have [MLflow Python SDK](https://pypi.org/project/mlflow/1.25.1/) installed on it and the environment variable should be set as:

```bash
MLFLOW_TRACKING_URI=http://mlflow-tracking-server.kubeflow
```

## Use MLflow

For more information on how to use MLflow Tracking, MLflow Projects, or MLflow Models, refer to the [MLflow][mlflow_docs] documentation.

If you want to see an example of how to log SDK metrics in MLflow, refer to the [Log and Hyperparameters Metrics to MLFlow][tutorial] tutorial.

[mlflow_docs]: https://mlflow.org/docs/1.25.1/index.html

> [tutorial]: https://docs.d2iq.com/dkp/kaptain/2.1.0/tutorials/sdk/quick-start#log-and-hyperparameters-metrics-to-mlflow

Author: ```suggestion
[tutorial]: ../sdk/quick-start#log-and-hyperparameters-metrics-to-mlflow
```
